chain/network/src/peer/peer_actor.rs

```diff
@@ -211,17 +221,11 @@ impl PeerActor {
     }
 
     fn send_message(&mut self, msg: &PeerMessage) -> Result<(), IOError> {
-        if let Some(enc) = self.force_encoding {
+        if let Some(enc) = self.encoding() {
             return self.send_message_with_encoding(msg, enc);
         }
-        if self.peer_status == PeerStatus::Connecting {
-            self.send_message_with_encoding(msg, Encoding::Proto)?;
-            self.send_message_with_encoding(msg, Encoding::Borsh)?;
-        } else if self.protocol_buffers_supported {
-            self.send_message_with_encoding(msg, Encoding::Proto)?;
-        } else {
-            self.send_message_with_encoding(msg, Encoding::Borsh)?;
-        }
+        self.send_message_with_encoding(msg, Encoding::Proto)?;
+        self.send_message_with_encoding(msg, Encoding::Borsh)?;
```

Comment:
the node should be able to gracefully handle garbage data from the peer, since it is an external source of data. We log parsing error as "debug", and definitely shouldn't log it as "error". What is the difference if we discard a message before or after decoding? IMO decoding a message that we intend to discard increases a risk of introducing a bug in which we start interpreting the message before discarding.

Reply:
Ok if we treat any de serialization errors as a pure debug info, then this seems okay.